### PR TITLE
Unit tests tested by PHPStan

### LINKED ISSUE #1122: `test` is not covered by static analysis
Discussed with @ondrejmirtes - `test/` is not covered by static analysis :scream: 

We should have both PHPStan and Psalm cover that ASAP, even if it means expanding the baseline by a lot.

### PR DESCRIPTION
**Pull request by kukulich**
Currently with generated baseline with 323 errors.

Fixes #1122

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,9 @@
     ],
     "require-dev": {
         "doctrine/coding-standard": "^9.0.0",
+        "phpstan/extension-installer": "^1.1.0",
```
Comment (Ocramius) [inline composer.json]: Can we install the extension without the extension installer?

File: phpstan.neon [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+
 parameters:
+    phpVersion: 80100
+
     level: 6
     checkGenericClassInNonGenericObjectType: false
 
     paths:
         - src
+        - test/unit
```
Comment (Ocramius) [inline phpstan.neon:15]: Let's also check `test/benchmark`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (kukulich) [reply composer.json]: Yes, done.

File: phpstan.neon (hunk 2 above)
Comment (kukulich) [reply phpstan.neon:15]: Done.

**Review by Ocramius — APPROVED** [review]

**Pull request merged**